larq/optimizers_v2.py

```diff
@@ -0,0 +1,82 @@
+import tensorflow as tf
+import numpy as np
+import larq as lq
+
+from larq import utils
+from copy import deepcopy
+
+
+@utils.register_keras_custom_object
+class Bop(tf.keras.optimizers.Optimizer):
```

Comment:
Could you add docstrings and a link to our paper similar to the xavier learning rate scaling callback?